ha/setup/create_cluster.py

```diff
@@ -0,0 +1,92 @@
+#!/bin/python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import argparse
+from ha.execute import SimpleCommand
+from cortx.utils.log import Log
+
+def cluster_auth(username, password, nodelist):
+    """Authorizes cluster nodes.
+
+    Parameters:
+            username - just a user name to make cluster authorization
+            password - just a password for aforementioned user
+            nodelist - List with nodes which to authorize
+    Returns: None."""
+    nodes = " ".join(nodelist)
+    cmd_auth = f"pcs cluster auth -u {username} -p {password} {nodes}"
+    SimpleCommand().run_cmd(cmd_auth)
+
+def cluster_create(cluster_name, nodelist, start=False, enable=True):
+    """Creates cluster on given nodes. Enables and starts cluster if needed.
+
+    Parameters:
+        cluster_name    - name of the cluster to be created
+        nodelist        - List with nodes where setup the cluster
+        start           - whether cluster shall be started immediately
+        enable          - whether cluster service shall start on boot
+
+    Returns: None."""
+    nodes = " ".join(nodelist)
+    cmd_setup = f"pcs cluster setup --name {cluster_name} {nodes}"
+    cmd_stonith = "pcs property set stonith-enabled=False"
+    cmd_start = f"pcs cluster start {nodes}"
+    cmd_enable = f"pcs cluster enable {nodes}"
+
+    cmdlist = [ cmd_setup, cmd_stonith ]
+    if enable:
+        cmdlist.append(cmd_enable)
+    if start:
+        cmdlist.append(cmd_start)
+    for s in cmdlist:
+        SimpleCommand().run_cmd(s)
```

Comment:
If the SimpleCommand fails in the last command, the function will execute the same commands again. Is it going to work?
I guess, simplest thing in one node cluster would be to destroy the cluster if any command fails and re-executing it will take care of that.

Reply:
If any command fails - exception is generated and execution is stopped. I don't think that we shall destroy cluster in the function which is supposed to create it. I would prefer the higher level to process such case  making decision to call cluster destroy if needed.
What we can do here - we can return adequate custom exception like `class ClusterSetupError(Exception)`, because upper level (ha_setup) is only interested in **the fact** of the failure, not the reason and HACommandTerminate (or even Exception) is to low-level information for it.

What do you think?

Reply:
Ok. Mini-provisioner will handle it.